### ISSUE TITLE
Failing unit tests on Erlang 19.1

### ISSUE THREAD

**Issue by aidecoe** [processone/p1_oauth2#1]
Unit tests fail because of the following warning:

```
src/oauth2_token.erl:68: Warning: crypto:rand_bytes/1 is deprecated and will be removed in a future release; use crypto:strong_rand_bytes/1
```

I have worked around it with:

```
sed -e '/, warnings_as_errors/d' -i rebar.test.config
```

but hopefully it's going to be fixed before next Erlang release when the function is removed.

**Comment by badlop** [comment]
From what I see, crypto:strong_rand_bytes/1 was added in 2011, OTP 14B03
- commit: https://github.com/erlang/otp/commit/f505644be64cd4844742b361a8ff411d70f4aa88
- release notes: http://erlang.org/download/otp_src_R14B03.readme

And it is surely available in Erlang/OTP 17.1, which is the oldest one required by recent ejabberd versions, so that function can be used in recent ejabberd.

This problem is also tracked in the original code repository, with good proposed fixes: https://github.com/kivra/oauth2/pull/68

According to the code, the old rand_bytes/1 function is used only when there isn't enough entropy, which should be a case rare enough to simply don't care about it. In that case, a stupid patch like this would apply:

```
diff --git a/src/oauth2_token.erl b/src/oauth2_token.erl
index 20fac2c..f5ec37e 100644
--- a/src/oauth2_token.erl
+++ b/src/oauth2_token.erl
@@ -54,19 +54,13 @@ is_alphanum(C) when C >= 16#61 andalso C =< 16#7A -> true;
 is_alphanum(_)                                    -> false.

 %% @doc Generate N random bytes, using the crypto:strong_rand_bytes
-%%      function if sufficient entropy exists. If not, use crypto:rand_bytes
-%%      as a fallback.
+%%      function if sufficient entropy exists.
 -spec rand_bytes(non_neg_integer()) -> binary().
 rand_bytes(N) ->
-    try
         %% NOTE: Apparently we can't meck away the crypto module,
         %% so we install this proxy to allow for testing the low_entropy
         %% situation.
-        ?MODULE:strong_rand_bytes_proxy(N)
-    catch
-        throw:low_entropy ->
-            crypto:rand_bytes(N)
-    end.
+        ?MODULE:strong_rand_bytes_proxy(N).

 %% @equiv crypto:strong_rand_bytes(N)
 -spec strong_rand_bytes_proxy(non_neg_integer()) -> binary().
```

**Comment by badlop** [comment]
I've applied some fixes from the upstream kivra repository, and one of them fix that: https://github.com/processone/p1_oauth2/commit/bd8c83ff4d689015d0453834164bb8722721c66a
